internal/channel/plugin.go

```diff
@@ -36,33 +36,49 @@ func NewPlugin(pluginType string, logger *zap.SugaredLogger) (*Plugin, error) {
 
 	cmd := exec.Command(file)
 
-	writer, err := cmd.StdinPipe()
+	var ioPipesToClose []io.Closer
+	defer func() {
+		for _, pipe := range ioPipesToClose {
+			_ = pipe.Close()
+		}
+	}()
+
+	reqRead, reqWrite, err := os.Pipe()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create stdin pipe: %w", err)
+		return nil, fmt.Errorf("failed to create reqRead/reqWrite pipe: %w", err)
 	}
+	ioPipesToClose = append(ioPipesToClose, reqRead)
 
-	reader, err := cmd.StdoutPipe()
+	resRead, resWrite, err := os.Pipe()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create stdout pipe: %w", err)
+		ioPipesToClose = append(ioPipesToClose, reqWrite)
+		return nil, fmt.Errorf("failed to create resRead/resWrite pipe: %w", err)
 	}
+	ioPipesToClose = append(ioPipesToClose, resWrite)
 
-	errPipe, err := cmd.StderrPipe()
+	logRead, logWrite, err := os.Pipe()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create stderr pipe: %w", err)
+		ioPipesToClose = append(ioPipesToClose, reqWrite, resRead)
+		return nil, fmt.Errorf("failed to create logRead/logWrite pipe: %w", err)
 	}
```

Comment:
In those two error cases, you're listing the names of files from all previous steps. Yes, it's only two times, so not a lot of repetition, but I quite like how `os/exec` does this internally, by tracking separately which ones are for the child process (those are closed always) and which ones are for the parent process (those are only closed in case of an error). That way, you only need to reference specific files in the block that's responsible for creating them.